robotPositioner/robotPositioner.h

```diff
@@ -0,0 +1,382 @@
+#include <vector>
+#include <list>
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <chrono>
+#include <csignal>
+#include <thread>
+#include <algorithm>
+
+
+// BoB robotics includes
+#include "../vicon/capture_control.h"
+#include "../vicon/udp.h"
+#include "../robots/norbot.h"
+#include "../video/see3cam_cu40.h"
+#include "../imgproc/opencv_unwrap_360.h"
+
+using namespace BoBRobotics::Vicon;
+using namespace BoBRobotics::ImgProc;
+using namespace BoBRobotics::Video;
+using namespace std::literals;
+
+#define PI 3.14159265359
+
+class robotPositioner {
+
+private:
+
+    // tracking and camera variables
+    std::list<std::vector<double>> m_coordinates;             // goal coordinates
+    UDPClient<ObjectData> *m_vicon;                           // vicon 
+    CaptureControl *m_viconCaptureControl;                    // vicon capture
+    BoBRobotics::Robots::Norbot *m_bot;                       // robot interface
+    See3CAM_CU40 *m_cam;                                      // camera
+    OpenCVUnwrap360 m_unwrapper;                              // image unwrapper
+    int m_imageNumber;                                        // image id number 
+    bool m_hasCamera;                                         // a boolean to indicate whether we have camera
+  
+    // Robot variables
+    double m_pos_X;                                           // robot's x position
+    double m_pos_Y;                                           // robot's y position
+    double m_pos_Z;                                           // robot's z position
+    double m_heading;                                         // heading (angle) of the robot
+    double m_goalPositionX;                                   // goal position x
+    double m_goalPositionY;                                   // goal position y
+    double m_goalPositionZ;                                   // goal position z
+    double m_bearingFromGoal;                                 // bearing (angle) from goal coordinate
+    double m_distanceFromGoal;                                // Euclidean distance from goal coordinate
+    double m_goalAngle;                                       // angle to turn after finding the correct location
+    double m_bearingFinal;                                    // heading converted to range <-180, 180> from <0, 360>
+
```

Comment:
It would be great if you could use the units library we're using for all these values -- it gives you c++ types for metres, angles etc.: https://github.com/nholthaus/units

You'd just need to change the types of the positions from double to millimeter_t and the angle variables to degree_t or radian_t. See e.g. here for example: https://github.com/BrainsOnBoard/bob_robotics/blob/master/vicon/udp.h

Reply:
...and you can make liberal use of the ``auto`` keyword with it to reduce typing

Reply:
there is a clear issue with this library when doing equations. I have w = k*v  where 'w' is angular velocity, 'k' is a constant and 'v' is velocity. It does not let me to calculate it because of type mismatch. I am not even surprised because k*v is still just a velocity which is not the type w needs to be

A possible workaround is that I am extracting the values and work with that, otherwise, I am getting all sorts of errors. I think this library is not able to deal with equations the way we expect it. Please double check and correct me if I am wrong.

Reply:
Well one solution would be to give k a unit in angle per metre (which is a sensible unit of turning speed)

Reply:
Yeah, Norbert just emailed me about this and I said the same thing. It does depend on whether w is a real physical unit though, rather than something like the turning speed passed into ``Tank.move()``.

But if v is a value in m/s and w is in deg/sec, then k will have a value in deg/m and you should be able to express it like this:
```
const auto k = 2_deg / 1_m; // for k==2
```